app/drivers/kscan/kscan_gpio_matrix.c

```diff
@@ -51,257 +159,301 @@ static int kscan_gpio_config_interrupts(const struct device **devices,
-                                                gpio_pin_get(in_dev, in_cfg->pin) > 0);            \
-            }                                                                                      \
-            err = gpio_pin_set(out_dev, out_cfg->pin, 0);                                          \
-            if (err) {                                                                             \
-                LOG_ERR("Failed to set output inactive (err %d)", err);                            \
-                return err;                                                                        \
-            }                                                                                      \
-        }                                                                                          \
-        /* Set all our outputs as active again. */                                                 \
-        COND_INTERRUPTS(kscan_gpio_set_output_state_##n(dev, 1);)                                  \
-        for (int r = 0; r < INST_MATRIX_ROWS(n); r++) {                                            \
-            for (int c = 0; c < INST_MATRIX_COLS(n); c++) {                                        \
-                bool pressed = read_state[r][c];                                                   \
-                /* Follow up reads needed because further interrupts won't fire on already tripped \
-                 * input GPIO pins */                                                              \
-                COND_INTERRUPTS(submit_follow_up_read = (submit_follow_up_read || pressed);)       \
-                if (pressed != data->matrix_state[r][c]) {                                         \
-                    LOG_DBG("Sending event at %d,%d state %s", r, c, (pressed ? "on" : "off"));    \
-                    data->matrix_state[r][c] = pressed;                                            \
-                    data->callback(dev, r, c, pressed);                                            \
-                }                                                                                  \
-            }                                                                                      \
-        }                                                                                          \
-        COND_INTERRUPTS(                                                                           \
-            if (submit_follow_up_read) {                                                           \
-                COND_CODE_0(DT_INST_PROP(n, debounce_period), ({ k_work_submit(&data->work); }),   \
-                            ({                                                                     \
-                                k_delayed_work_cancel(&data->work);                                \
-                                k_delayed_work_submit(&data->work, K_MSEC(5));                     \
```

Comment:
Was this an error in the original code? This is conditional based on there being a debounce period, but then it uses a hard-coded 5 ms instead of the debounce period. I added a separate DT property to control this, but if this was supposed to be the debounce time I can remove it.

Reply:
I removed the separate property assuming that this was intended to use the debounce period.

Reply:
Yeah, this was a bug.